ValveKeyValue/ValveKeyValue.Test/Text/BaseIncludeTestCase.cs

```diff
@@ -42,8 +42,18 @@ sealed class StubIncludedFileLoader : IIncludedFileLoader
         {
             Stream IIncludedFileLoader.OpenFile(string filePath)
             {
-                Assert.That(filePath, Is.EqualTo("file.vdf"));
-                return TestDataHelper.OpenResource("Text.kv_base_included.vdf");
+                if (filePath == "file.vdf")
+                {
+                    return TestDataHelper.OpenResource("Text.kv_base_included.vdf");
+                }
+                else if (filePath == "this_file_does_not_exist.vdf")
+                {
+                    return Stream.Null;
```

Comment:
So far we haven't defined any contract for how an `IIncludedFileLoader` should behave if it can't achieve the goal of returning a `Stream` that represents the file.

I would have expected either to return `null` (i.e. make the contract `Stream?`), or to throw a standard exception that we then handle (such as `FileNotFoundException` or `DirectoryNotFoundException`). This appears to do neither and returns an empty stream, which I don't think is an appropriate solution.

I'd also be tempted to make this 'new' behaviour optional (though perhaps default-on), so that consumers can decide if they want invalid references to fail or if they want the Valve behaviour of swallowing the error and continuing anyway.

Reply:
Return null to swallow, throw an exception if you want an error?

Sounds convoluted but achieves both points.

Reply:
I've implemented xPaw's suggestion for now. Let me know if this satisfies the criteria.